doc/examples/transform/plot_tps_deformation.py

```diff
@@ -0,0 +1,129 @@
+r"""
+==========================================
+Interpolate images with thin-plate splines
+==========================================
+
+Thin-plate splines (TPS) refer to a method for interpolating data [1]_.
+In an image context, we use a 2D generalization of TPS, i.e., a two-dimensional
+simulation of a 1D cubic spline, which is the solution of the biharmonic equation [2]_.
+According to [3]_, the basic solution of the biharmonic function was expanded.
+Given corresponding source and target control points, TPS is used to transform a space.
+In this case our space is a 2D image.
+
+It should be noted that the commonly used in image processing
+Affine Transformation [4]_ can be understood as special variant of TPS.
+
+This script shows how to use the Thine Plate Spine Transformation for
+image warping.
+
+For further information on TPS Transformation, see:
+
+.. [1] Wikipedia, Thin plate spline
+       https://en.wikipedia.org/wiki/Thin_plate_spline
+
+.. [2] Weisstein, Eric W. "Biharmonic Equation." From MathWorld--A Wolfram Web Resource
+       https://mathworld.wolfram.com/BiharmonicEquation.html
+
+.. [3] Bookstein, Fred L. "Principal warps: Thin-plate splines and the
+       decomposition of deformations." IEEE Transactions on pattern analysis and
+       machine intelligence 11.6 (1989): 567–585.
+
+.. [4] Wikipedia, Affine transformation
+       https://en.wikipedia.org/wiki/Affine_transformation#Image_transformation
+
+
+Image Interpolation
+===================
+
+In this example we will see how to use thin plate spline interpolation in the
+context of interpolating the correspondence of five pairs of landmarks.
+The examples shows landmarks before and after being bent using the TPS algorithm,
+and the control points (X) used as input the the algorithm.
+"""
+import matplotlib.pyplot as plt
+import numpy as np
+
+import skimage as ski
+
+chess = ski.data.checkerboard()
+
+src = np.array([[3.6929, 10.3819],[6.5827, 8.8386], [6.7766, 12.0866], [4.8189, 11.2047], [5.6969, 10.0748]])
+dst = np.array([[3.9724, 6.5354], [6.6969, 4.1181], [6.5394, 7.2362], [5.4016, 6.4528], [5.7756, 5.1142]])
+
+src *= chess.shape[0]//15
+dst *= chess.shape[0]//15
+# Fit the thin plate spline from output to input
+tps = ski.transform.TpsTransform()
+warped_img = ski.transform.tps_warp(chess, src, dst, grid_scaling=1)
+
+
+fig, axs = plt.subplots(1, 2, figsize=(16, 8))
+# axs[0].axis('off')
+# axs[1].axis('off')
+
+labels = ['1', '2', '3', '4', '5']  # Adjust the number of labels to match the number of points
+
+axs[0].imshow(chess[..., ::-1], origin='upper', cmap='gray')
+axs[0].scatter(src[:, 0], src[:, 1] , marker='x', color='green')
+
+for i, label in enumerate(labels):
+    axs[0].annotate(label, (src[:, 0][i], src[:, 1][i] ),
+                    textcoords="offset points", xytext=(0, 10), ha='center', color='red')
+
+axs[1].imshow(warped_img[..., ::-1], origin='upper', cmap='gray')
+axs[1].scatter(dst[:, 0] , dst[:, 1] , marker='x', color='green')
```

Comment:
Something about this result seems off to me. In https://output.circle-artifacts.com/output/job/56bdac54-7c2b-425b-909d-e54f7a7bac09/artifacts/0/doc/build/html/auto_examples/transform/plot_tps_deformation.html#sphx-glr-auto-examples-transform-plot-tps-deformation-py it doesn't look like the warped chess grid and the points go together. The points should still be at the "same position relative" to the chess grid.

![image](https://github.com/scikit-image/scikit-image/assets/20140352/e235105a-e484-45fb-9983-f50f1c45fac8)

Reply:
Maybe the choice of landmarks isn't ideal for this example. In my views, the points shouldn't be at the same position relative to the chess grid but shift to the target points relative to the chess grid. I updated the gallery example to use the astronaut image.